corehq/apps/app_manager/suite_xml/xml_models.py

```diff
@@ -86,21 +85,28 @@ class Id(XmlObject):
 
 class XPathEnum(TextXPath):
     @classmethod
-    def build(cls, enum, template, get_template_context, get_value):
+    def build(cls, enum, type, template, get_template_context, get_value):
+        xpath_template = template[0]
+        format = template[1]
+        print(template)
         variables = []
         for item in enum:
             v_key = item.key_as_variable
             v_val = get_value(v_key)
             variables.append(XPathVariable(name=v_key, locale_id=v_val))
-
         parts = []
         for i, item in enumerate(enum):
             template_context = get_template_context(item, i)
-            parts.append(template.format(**template_context))
-        parts.append("''")
-        parts.append(")" * len(enum))
-        function = ''.join(parts)
+            parts.append(xpath_template.format(**template_context))
+        if type == "display" and format == "enum":
+            parts.insert(0, "replace(join(' ', ")
+            parts[-1] = parts[-1][:-2] # removes extra comma from last string
+            parts.append("), '  ', '')")
```

Comment:
What will this do with three consecutive spaces? I'd guess you want to match on ` \s+` to get any number of consecutive spaces.